src/app/state/actions/index.tsx

```diff
@@ -1231,6 +1231,18 @@ export const adminModifyUserEmailVerificationStatuses =
     }
   };
 
+export const adminModifyTeacherPending = (status: boolean, userIds: number[]) => async (dispatch: Dispatch<Action>) => {
+  dispatch({ type: ACTION_TYPE.ADMIN_MODIFY_TEACHER_PENDING_REQUEST });
+  try {
+    const response = await api.admin.modifyTeacherPending.post(status, userIds);
+    dispatch({ type: ACTION_TYPE.ADMIN_MODIFY_TEACHER_PENDING_RESPONSE_SUCCESS });
+    return response.data;
```

Comment:
This is not how you're supposed to do it with Redux. If you check the other functions in this file, you'll see that we're not returning data directly from the function. Instead, you dispatch it (you need to pass it as part of the object that is the parameter of `dispatch`). You should double-check this, though, because it's been a long time since I've used Redux.

Reply:
I've changed this now, adding a useEffect to check for the value in state. I've not used redux very much (except making small modifications on this project!) but I think it's correct now. It works at least!